src/Microsoft.Extensions.Configuration.UserSecrets/ConfigurationExtensions.cs

```diff
@@ -14,8 +13,13 @@ public static class ConfigurationExtensions
         /// </summary>
         /// <param name="configuration"></param>
         /// <returns></returns>
-        public static IConfigurationBuilder AddUserSecrets([NotNull]this IConfigurationBuilder configuration)
+        public static IConfigurationBuilder AddUserSecrets(this IConfigurationBuilder configuration)
         {
+            if (configuration == null)
+            {
+                throw new ArgumentNullException(nameof(configuration));
```

Comment:
The parameter name isn't `this`, it's `configuration`. The `this` is just syntactic sugar for the compiler.